src/MCPServer/lib/linkTaskManagerChoice.py

```diff
@@ -137,10 +138,10 @@ def xmlify(self):
         etree.SubElement(ret, "UUID").text = self.jobChainLink.UUID
         ret.append(self.unit.xmlify())
         choices = etree.SubElement(ret, "choices")
-        for chainAvailable, description in self.choices:
+        for id_, description, __ in self.choices:
             choice = etree.SubElement(choices, "choice")
-            etree.SubElement(choice, "chainAvailable").text = chainAvailable.__str__()
-            etree.SubElement(choice, "description").text = description
+            etree.SubElement(choice, "chainAvailable").text = id_
+            etree.SubElement(choice, "description").text = text_type(description)
```

Comment:
There are some bits in this commit that should be long to "Use new workflow in MCPServer engine", like this one above. These are just small fixes I had to apply down the road while I was doing testing and for some reason they ended up in this other commit.